src/main/java/ch/njol/skript/expressions/ExprElement.java

```diff
@@ -124,54 +175,74 @@ public <R> Expression<? extends R> getConvertedExpression(Class<R>... to) {
 		if (convExpr == null)
 			return null;
 
-		ExprElement exprElement = new ExprElement();
-		exprElement.type = this.type;
+		ExprElement<R> exprElement = new ExprElement<>();
 		exprElement.expr = convExpr;
-		exprElement.number = this.number;
-		return (Expression<? extends R>) exprElement;
+		exprElement.startIndex = startIndex;
+		exprElement.endIndex = endIndex;
+		exprElement.type = type;
+		return exprElement;
 	}
 
 	@Override
 	public boolean isSingle() {
-		return true;
+		return type != ElementType.FIRST_X_ELEMENTS && type != ElementType.LAST_X_ELEMENTS && type != ElementType.RANGE;
+	}
+
+	@SuppressWarnings("unchecked")
+	private Class<T> getReturnType0() {
```

Comment:
Can we have a better method naming like getReturnType_i or get getClassReturnType or getReturnTypeClass?